icu4c/source/common/mlbe.cpp

```diff
@@ -128,135 +135,127 @@ int32_t MlBreakEngine::divideUpRange(UText *inText, int32_t rangeStart, int32_t
+                                          int32_t numCodeUnits, int32_t numBreaks,
+                                          UVector32 &boundary, UErrorCode &status) const {
     if (U_FAILURE(status)) {
-        return;
+        return numBreaks;
     }
-
-    UnicodeString feature;
+    int32_t start = 0, end = 0;
     int32_t score = fNegativeSum;
 
-    if (elementList[0] != INVALID) {
-        // When the key doesn't exist, Hashtable.geti(key) returns 0  and 2 * 0 = 0.
-        // So, we can skip to check whether fModel includes key featureList[j] or not.
-        score += (2 * fModel.geti(feature.setTo(u"UW1:", 4).append(elementList[0])));
-    }
-    if (elementList[1] != INVALID) {
-        score += (2 * fModel.geti(feature.setTo(u"UW2:", 4).append(elementList[1])));
-    }
-    if (elementList[2] != INVALID) {
-        score += (2 * fModel.geti(feature.setTo(u"UW3:", 4).append(elementList[2])));
-    }
-    if (elementList[3] != INVALID) {
-        score += (2 * fModel.geti(feature.setTo(u"UW4:", 4).append(elementList[3])));
-    }
-    if (elementList[4] != INVALID) {
-        score += (2 * fModel.geti(feature.setTo(u"UW5:", 4).append(elementList[4])));
-    }
-    if (elementList[5] != INVALID) {
-        score += (2 * fModel.geti(feature.setTo(u"UW6:", 4).append(elementList[5])));
-    }
-    if (elementList[1] != INVALID && elementList[2] != INVALID) {
-        score += (2 * fModel.geti(
-                          feature.setTo(u"BW1:", 4).append(elementList[1]).append(elementList[2])));
-    }
-    if (elementList[2] != INVALID && elementList[3] != INVALID) {
-        score += (2 * fModel.geti(
-                          feature.setTo(u"BW2:", 4).append(elementList[2]).append(elementList[3])));
-    }
-    if (elementList[3] != INVALID && elementList[4] != INVALID) {
-        score += (2 * fModel.geti(
-                          feature.setTo(u"BW3:", 4).append(elementList[3]).append(elementList[4])));
-    }
-    if (elementList[0] != INVALID && elementList[1] != INVALID && elementList[2] != INVALID) {
-        score += (2 * fModel.geti(feature.setTo(u"TW1:", 4)
-                                      .append(elementList[0])
-                                      .append(elementList[1])
-                                      .append(elementList[2])));
-    }
-    if (elementList[1] != INVALID && elementList[2] != INVALID && elementList[3] != INVALID) {
-        score += (2 * fModel.geti(feature.setTo(u"TW2:", 4)
-                                      .append(elementList[1])
-                                      .append(elementList[2])
-                                      .append(elementList[3])));
+    for (int i = 0; i < 6; i++) {
+        // UW1 ~ UW5
+        start = startIdx + i;
+        if (indexList[start] != -1) {
+            end = (indexList[start + 1] != -1) ? indexList[start + 1] : numCodeUnits;
+            score +=
+                fModel[i].geti(inString.tempSubString(indexList[start], end - indexList[start]));
```

Comment:
I think the  fModel[i] (in line 154)  fModel[i + 6] (in line 162) and  fModel[i + 9] (in line 171) is too hard to read
should we add an internal enum in this file such as 
```
enum class ModelIndex {
  kUWStart = 0,
  kBWStart = 6,
  kTWStart = 9
};
```
and use
```
 fModel[ModelIndex::kUWStart + i].geti(....)
 fModel[ModelIndex::kBWStart + i].geti(....)
 fModel[ModelIndex::kTWStart + i].geti(....)
```
?

Reply:
Done